automation_testing/pages/Form/FormWithLayoutPage.ts

```diff
@@ -0,0 +1,91 @@
+import { expect, Locator, Page } from "@playwright/test";
+import { BasePage } from "../BasePage";
+
+export class FormWithLayout extends BasePage {
+
+	readonly page: Page;
+	readonly topComponentContainer: Locator;
+	readonly topComponentContainerSections: Locator;
+	readonly sectionContainer: Locator;
+	readonly sectionRows: Locator;
+
+	constructor(page: Page) {
+		super(page);
+		this.page = page;
+		this.topComponentContainer = page.locator("//*[@id='root']/div/form/div[1]/div[2]");
+		this.topComponentContainerSections = page.locator("a");
+		this.sectionContainer = page.locator("//*[@id='root']/div/form/div[2]/div");
+		this.sectionRows = page.locator("//*[@id='root']/div/form/div[2]/div[1]/div/div");
+	}
+
+	async getNumberOfSectionsFromTopComponent():Promise<number> {
+		await this.topComponentContainer.waitFor();
+		return await this.topComponentContainerSections.count();
+	}   
+
+	async getNumberOfSections():Promise<number> {
+		return await this.sectionContainer.count();
+	} 
+
+	async getSectionsTitlesFromTopComponent():Promise<string[]> {
+		const numberOfSections = await this.getNumberOfSectionsFromTopComponent();
+		const sectionTitles = [];
+		for (let i = 0; i <  numberOfSections; i++) {
+			sectionTitles.push(await this.topComponentContainer.locator("a").nth(i).textContent());
+		}
+		return sectionTitles;
+	}
+	
+	async getPositionOfSection(section:string):Promise<number> {
+		const sections = this.getSectionsTitlesFromTopComponent();
+		return (await sections).indexOf(section);
+	} 
+
+	async getSectionsTitles():Promise<string[]> {
+		const numberOfSections = await this.getNumberOfSectionsFromTopComponent();
+		const sectionTitles = [];
+		for (let i = 0; i <  numberOfSections; i++) {
+			sectionTitles.push(await this.sectionContainer.locator("h1").nth(i).textContent());
+		}
+		return sectionTitles;
+	}
+
+	async getSectionsDescriptions():Promise<string[]> {
```

Comment:
I feel the code in this function is repeated a lot and we have an area of opportunity if we make it generic, I feel instead of having the functions getSectionsTitlesFromTopComponent, getSectionsTitles, getSectionsDescriptions, and getNumberOfColumsInEachRowFromSections we could have something like getSectionsData that receives as argument the type of data to find

For example:
```
	async getSectionsData(container: locator, element: string):Promise<string[]> {
                   ...all your logic
                   for loop
                           container.locator(element)...
        }
```

That way we can make this work with whatever might change / be added in the future.

Reply:
Created getSectionData method to address this.